From: Sisyphus <sisyphus@ohmyopencode.com>
Date: Wed, 20 May 2026 19:30:00 -0600
Subject: [PATCH] driver/razerkbd: fix hid_report_raw_event 6-argument signature

Kernel commit 0a3fe972a7cb ("HID: pass the buffer size to
hid_report_raw_event") changed the function signature from 5 to 6
parameters, adding size_t bufsize between u8 *data and u32 size.

Conditionally pass the 6th argument on kernels >= 7.0.0
(LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)).

Signed-off-by: Sisyphus <sisyphus@ohmyopencode.com>

--- a/driver/razerkbd_driver.c
+++ b/driver/razerkbd_driver.c
@@ -34,6 +34,8 @@
 #include "razercommon.h"
 #include "razerchromacommon.h"
 
+#include <linux/version.h>
+
 /*
  * Version Information
  */
@@ -4829,12 +4831,22 @@ static int razer_raw_event_bitfield(struct hid_device *hdev, struct razer_kbd_de
 			if (report_extra) {
 				u8 xdata[22] = { 0x02 };
 
 				// report key down
 				xdata[1] = cur_value;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+				hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), sizeof(xdata), 0);
+#else
 				hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
+#endif
 
 				// report key up
 				xdata[1] = 0x00;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+				hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), sizeof(xdata), 0);
+#else
 				hid_report_raw_event(hdev, HID_INPUT_REPORT, xdata, sizeof(xdata), 0);
+#endif
 			}
 		}
 	}
